services/QuillLMS/app/models/concerns/student.rb

```diff
@@ -67,9 +67,8 @@ def move_activity_sessions(old_classroom, new_classroom)
           hide_extra_activity_sessions(cu.id)
         end
       else
-
         new_unit_name = "#{name}'s Activities from #{old_classroom.name}"
-        unit = Unit.create(user_id: new_classroom.owner.id, name: new_unit_name)
+        unit = Unit.create_with_incremented_name(user_id: new_classroom.owner.id, name: new_unit_name)
         new_cu = ClassroomUnit.find_or_create_by(unit_id: unit.id, classroom_id: new_classroom_id, assigned_student_ids: [user_id])
```

Comment:
Should all 20 similar names be already taken for some reason, this line will raise a RuntimeError

Reply:
No way around this edge case  - alternatively we could append SecureRandom.id to the name and remove the retry logic.